remoteappmanager/cli/remoteapprest/__main__.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+import sys
+import os
+import requests
+import requests.utils
+import json
+from urllib.parse import urljoin
+
+import click
+
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from requests.packages.urllib3 import disable_warnings
+
+
+class Credentials:
+    """Data class to hold the credentials extracted from the credential file.
+    """
+    def __init__(self, url, username, cookies):
+        self.url = url
+        self.username = username
+        self.cookies = cookies
+
+    @classmethod
+    def from_file(cls, credentials_file):
+        """Extracts the authorization info from the credentials file.
+        Returns a tuple with url, username, and a dict of credentials
+        cookies"""
+        with open(credentials_file, "r") as f:
+            lines = f.readlines()
+
+        url = lines[0].strip()
+        username = lines[1].strip()
+        cookies = {}
+        for line in lines[2:]:
+            k, v = line.split('=', 1)
+            cookies[k.strip()] = v.strip()
+
+        return cls(url, username, cookies)
+
+    def write(self, credentials_file):
+        """Stores the credentials in a credentials file."""
+        with open(credentials_file, "w") as f:
+            f.write("{}\n".format(self.url))
+            f.write("{}\n".format(self.username))
+            for k, v in self.cookies.items():
+                f.write("{}={}\n".format(k, v))
+
+
+class RemoteAppRestContext:
+    """The click context passed around."""
+    credentials_file = None
+    credentials = None
+
+
+@click.group()
+@click.option("--credentials-file",
+              default=os.path.expanduser("~/.remoteapprest"),
```

Comment:
Just thinking maybe we should have a directory to hold these kind of files "~/.remoteapp/.rest-credentials"